app/services/platform-apps/api/modules/scene-transitions.ts

```diff
@@ -0,0 +1,161 @@
+import * as path from 'path';
+import { getType } from 'mime';
+import { apiMethod, EApiPermissions, IApiContext, Module } from './module';
+import {
+  ETransitionType,
+  ITransitionCreateOptions,
+  TransitionsService,
+} from 'services/transitions';
+import { Inject } from 'util/injector';
+import { PlatformAppsService } from '../../index';
+
+type AudioFadeStyle = 'fadeOut' | 'crossFade';
+
+enum ObsAudioFadeStyle {
+  FadeOut = 0,
+  CrossFade = 1,
+}
+
+type TransitionPointType = 'time' | 'frame';
+
+enum ETransitionPointType {
+  Time = 0,
+  Frame = 1,
+}
+
+type StingerTransitionOptions = {
+  /** Transition type **/
+  type: 'stinger';
+  /** The name of the transition **/
+  name: string;
+  /** A relative URL to a video asset inside a Platform app **/
+  url: string;
+  /** How the audio should fade: fade out or crossfade. **/
+  audioFadeStyle?: AudioFadeStyle;
+  /**
+   * If set to true scene, the scene transition won't be editable by
+   * end-users after being created.
+   */
+  shouldLock?: boolean;
+  /** Whether to monitor audio **/
+  shouldMonitorAudio?: boolean;
+  /** A transition point in milliseconds **/
+  transitionPoint?: number;
+  /** Type of transition point, frame or time **/
+  transitionPointType?: TransitionPointType;
+};
+
+type TransitionOptions = StingerTransitionOptions;
```

Comment:
I'm curious why this and `StingerTransitionOptions` are written as types instead of interfaces.  I think I see what you're trying to do, but I think this would make much more sense if `TransitionOptions` were an interface, and then `StingerTransitionOptions` extended that interface.

Reply:
TransitionOptions is a sum type. The fact that we only support stinger transitions at the moment doesn't make it obvious, but I believe we've talked about how we'll differentiate based on the `type` property.